vendor-extra/ContentPageBundle/src/Event/CreateContentPageEvent.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentPageBundle\Event;
+
+use FluentDOM\DOM\Document;
+use Symfony\Component\EventDispatcher\Event;
+
+final class CreateContentPageEvent extends Event
+{
+    public const NAME = 'libero.page.content';
+
+    private $content = [];
+    private $context;
+    private $item;
+    private $title;
+
+    public function __construct(Document $item, array $context = [])
+    {
+        $this->item = $item;
+        $this->context = $context;
+    }
+
+    public function getContent() : array
+    {
+        return $this->content;
+    }
+
+    public function addContent($content) : void
+    {
+        $this->content[] = $content;
```

Comment:
this mutability stands in constrant with ordinary usage of immutable events, I'm interested in possible ways of separating the propagation of the event from a collecting the `content` being produced by listeners. Still wouldn't want a global service collecting everything as that has an even larger scope of mutability.

Reply:
Symfony events are really hooks.